frontend/src/metabase/admin/settings/notifications/WebhookForm.tsx

```diff
@@ -0,0 +1,212 @@
+import type { FormikHelpers } from "formik";
+import { t } from "ttag";
+import * as Yup from "yup";
+
+import { useTestChannelMutation } from "metabase/api/channel";
+import {
+  Form,
+  FormChipGroup,
+  FormProvider,
+  FormSubmitButton,
+  FormTextInput,
+} from "metabase/forms";
+import { FormKeyValueMapping } from "metabase/forms/components/FormKeyValueMapping";
+import { useActionButtonLabel } from "metabase/hooks/use-action-button-label";
+import { Button, Chip, Flex, Alert, Text, Group, Icon } from "metabase/ui";
+import type { NotificationAuthMethods } from "metabase-types/api";
+
+const validationSchema = Yup.object({
+  url: Yup.string()
+    .url(t`Please enter a correctly formatted URL`)
+    .required(),
+  name: Yup.string().required(t`Please add a name`),
+  description: Yup.string().required(t`Please add a description`),
+  "auth-method": Yup.string()
+    .required()
+    .equals(["none", "header", "query-param", "request-body"]),
+  "auth-info": Yup.object(),
+});
+
+const styles = {
+  input: {
+    fontWeight: 400,
+  },
+  label: {
+    fontSize: "14px",
+    marginBottom: "0.75rem",
+  },
+};
+
+export type WebhookFormProps = {
+  url: string;
+  name: string;
+  description: string;
+  "auth-method": NotificationAuthMethods;
+  "auth-info": Record<string, string>;
+};
+
+type WebhookFormikHelpers = FormikHelpers<WebhookFormProps>;
+
+export const handleFieldError = (e: any) => {
+  if (!e.data) {
+    return;
+  } else if (typeof e.data === "string") {
+    throw { data: { errors: { url: e.data } } };
+  } else if (e.data.message) {
+    throw { data: { errors: { url: e.data.message } } };
+  } else if (typeof e.data.errors === "object") {
+    throw e;
+  }
+};
```

Comment:
We have a `getResponseErrorMessage` function that does something similar, might be possible to re-use it